src/modal/Modal.tsx

```diff
@@ -167,8 +167,12 @@ const Modal: React.FC = () => {
               <button className="left-button" onClick={handleDismissClick}>
                 Dismiss
               </button>
-              <button className="right-button" onClick={handleConnectClick}>
-                Connect
+              <button
+                className="right-button"
+                onClick={handleConnectClick}
+                disabled={state.loading}
+              >
+                {state.loading ? "Loading..." : "Connect"}
```

Comment:
We should just use `useState` inside the component for whether we're loading. It's purely UI state. We should also disable the dismiss button and make the inputs read-only

Reply:
Resolved issues and committed